### PR TITLE
 set the value of cmi.mode

### PR DESCRIPTION
**Pull request by rimsha-zaib**

### CODE REVIEW

**Comment by rimsha-zaib** [comment]
@ziafazal Could you please review this PR?

**Comment by rimsha-zaib** [comment]
In this commit, I addressed the issue related to SCORM attempts not being set correctly. [https://github.com/orgs/overhangio/projects/4/views/1?pane=issue&itemId=37732590](url).
Upon review, it was found that the cmi.mode value was not being set appropriately. This caused discrepancies in SCORM attempts, leading to unexpected behavior.

**Comment by regisb** [comment]
Great explanation! You should add this to your git commit description.

File: openedxscorm/static/js/src/scormxblock.js [hunk 1]
```diff
@@ -177,27 +177,27 @@ function ScormXBlock(runtime, element, settings) {
...
+            var response = $.ajax({
                 type: "POST",
                 url: getValueUrl,
                 data: JSON.stringify({
-                    'name': cmi_element
+                    'name': cmi_element,
+                    'url': window.location.href
                 }),
-                async: false,
-                success: function (response) {
```
Comment (ziafazal) [inline openedxscorm/static/js/src/scormxblock.js:194]: why we have to remove this `success` handler?

File: openedxscorm/scormxblock.py [hunk 2]
```diff
@@ -406,13 +406,21 @@ def extract_folder_base_path(self):
...
+        return "normal"
 
     @XBlock.json_handler
     def scorm_get_value(self, data, _suffix):
         """
         Here we get only the get_value events that were not filtered by the LMSGetValue js function.
         """
         name = data.get("name")
+        if name in ["cmi.core.lesson_mode", "cmi.mode"]:
+            mode = self.get_mode(data)
```
Comment (ziafazal) [inline openedxscorm/scormxblock.py:422]: nit: can merge these two lines to simplify these
```suggestion
            return {"value": self.get_mode(data)}
```

**Review by ziafazal — COMMENTED** [review]
@rimsha-zaib overall look. I have a question and a suggestion. Just wondering is there any way we can get hostname on server side e.g `self.runtime.hostname` instead of sending it from javascript?

File: openedxscorm/static/js/src/scormxblock.js (hunk 1 above)
Comment (rimsha-zaib) [reply openedxscorm/static/js/src/scormxblock.js:194]: Unable to retrieve user mode through **self.runtime**. Although runtime has a **handleurl** function , isn't returning correct URLs.

Comment (rimsha-zaib) [reply openedxscorm/static/js/src/scormxblock.js:194]: Synchronous AJAX call within GetValue does not properly return the response value in success handler, resulting in an empty string. This causes GetValue to be called twice, so the mode is not being set in this case.

Comment (ziafazal) [reply openedxscorm/static/js/src/scormxblock.js:194]: I'm not getting it why synchronous AJAX call within GetValue does not properly return the response value?

**Review by ziafazal — APPROVED** [review]
👍

**Pull request merged**